frontend/catalyst/jit.py

```diff
@@ -591,28 +622,78 @@ def closure(qnode, *args, **kwargs):
             (qml.QNode, "__call__", closure),
         ):
             # TODO: improve PyTree handling
-
-            def fn_with_transform_named_sequence(*args, **kwargs):
-                """
-                This function behaves exactly like the user function being jitted,
-                taking in the same arguments and producing the same results, except
-                it injects a transform_named_sequence jax primitive at the beginning
-                of the jaxpr when being traced.
-
-                Note that we do not overwrite self.original_function and self.user_function;
-                this fn_with_transform_named_sequence is ONLY used here to produce tracing
-                results with a transform_named_sequence primitive at the beginning of the
-                jaxpr. It is never executed or used anywhere, except being traced here.
-                """
-                _inject_transform_named_sequence()
-                return self.user_function(*args, **kwargs)
-
             jaxpr, out_type, treedef = trace_to_jaxpr(
-                fn_with_transform_named_sequence, static_argnums, abstracted_axes, full_sig, kwargs
+                fn_with_transform_named_sequence,
+                static_argnums,
+                abstracted_axes,
+                full_sig,
+                kwargs,
             )
 
         return jaxpr, out_type, treedef, dynamic_sig
 
+    @instrument(size_from=0)
+    @debug_logger
+    def experimental_capture(self, args, **kwargs):
```

Comment:
I guess this is not very well known yet, but we should try to keep the structure/interface of the QJIT class stable unless we really have to change it, as it provides an interface for other integrations. The existing stages "pre_processing, capture, ir generation, compilation, execution" are meant to be swappable without changing anything about the general process of the QJIT compilation.

Instead of creating an additional stage for the experimental capture and dispatching to it from the `jit_compile` function, we should dispatch _inside_ our implementation of the `capture` stage, which maintains the modularity.
The code here can be moved to a free-standing function somewhere (for example where the pl converter already lives if appropriate).

Reply:
I am now calling the experimental function from the capture one. Let me know if it is good?